### DISCUSSION TITLE
Creating a node from a .net SDK of a camera

### DISCUSSION THREAD

**Discussion by benefron** [bonsai-rx/bonsai#1259] [Q&A]
Hi,
I am trying to modify the camera capture node to allow me to use some of the functionality of my camera from it's .net SDK.
I am using imaging source cameras.
I am currently stuck at the level of the extensions.
I define in the extensions.csproj:
<Reference Include="TIS.Imaging">
      <HintPath>C:\Users\lampl\Documents\IC Imaging Control 3.5\redist\dotnet\x64\TIS.Imaging.ICImagingControl35.dll</HintPath>
</Reference>
for the dll.
And in the node itself I added: 
using TIS.Imaging;

But I get an error when running Bonsai:
e:\Ben\Bonsai test\Extensions\CameraCaptureWithIndex.cs(6,7) : error CS0246: The type or namespace name 'TIS' could not be found (are you missing a using directive or an assembly reference?)

Also is there a simple way to create from the SDK a package to use the full functionality of the camera SDK?

I don't have any background programing in C#

Thank you

**Comment by benefron** [comment]
Alternatively I also thought of creating a custom C# source node that uses cameraCapture and output from it the capture property assignment FrameCount. Is it possible? I am unable to get a node to output anything but the IplImage.

**Comment by RoboDoig** [comment]
Hi @benefron - If you are looking to integrate your camera SDK into Bonsai my suggestion would be to generate a new Bonsai package rather than do it by extensions, you can find instructions [here](https://bonsai-rx.org/docs/articles/create-package.html). Once you have your Visual Studio project set up, you can add TIS.Imaging.ICImagingControl35.dll as a dependency.

**Comment by benefron** [comment]
I really don't know how to do it. Is there a way to inherit the cameracapture node and get the metadata for each frame from the camera SDK? It has a class to get metadata from, but I need help with C#. I want to expose the metadata and captured frames which work excellently with the current cameraCapture node.
